### PR TITLE
Account balance

### PR DESCRIPTION
**Pull request by eesmerdag**

### CODE REVIEW

File: test/integration/api_test.go [hunk 1]
```diff
@@ -245,3 +244,59 @@ func TestGetDelegatorReward(t *testing.T) {
...
+			resp, err := capi.GetAccountBalance(ctx, tt.args)
+
+			if tt.wantErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, resp.Balances[0].Text, tt.resText)
+				require.Equal(t, resp.Balances[0].Currency, tt.resCurrency)
+				var n *big.Int
+				require.Equal(t, resp.Balances[0].Numeric, n)    // not available for cosmos
```
Comment (pacmessica) [inline test/integration/api_test.go:297]: lol this is available :p please add numeric

File: test/integration/api_test.go [hunk 2]
```diff
@@ -245,3 +244,59 @@ func TestGetDelegatorReward(t *testing.T) {
...
+				require.Equal(t, resp.Balances[0].Exp, int32(0)) // not available for cosmos
```
Comment (pacmessica) [inline test/integration/api_test.go:298]:
```suggestion
				require.Equal(t, resp.Balances[0].Exp, int32(0))
```

exp should be 0

File: client/client.go [hunk 3]
```diff
@@ -231,9 +238,50 @@ func (ic *IndexerClient) GetBlock(ctx context.Context, tr cStructs.TaskRequest,
...
+	}
+	close(out)
+
+	sendResp(ctx, tr.Id, out, ic.logger, stream, nil)
+}
+
 // GetReward gets reward
 func (ic *IndexerClient) GetReward(ctx context.Context, tr cStructs.TaskRequest, stream *cStructs.StreamAccess, client LCD) {
-	timer := metrics.NewTimer(getBlockDuration)
+	timer := metrics.NewTimer(getRewardDuration)
```
Comment (pacmessica) [inline client/client.go:284]: 👍

File: client/client.go [hunk 4]
```diff
@@ -231,9 +238,50 @@ func (ic *IndexerClient) GetBlock(ctx context.Context, tr cStructs.TaskRequest,
...
+	if err != nil {
+		stream.Send(cStructs.TaskResponse{
+			Id:    tr.Id,
+			Error: cStructs.TaskError{Msg: "Cannot unmarshal payload"},
+			Final: true,
+		})
+		return
+	}
+
+	sCtx, cancel := context.WithTimeout(ctx, time.Second*2)
```
Comment (pacmessica) [inline client/client.go:257]: can you make this a constant at the top of the file eg `ctxTimeout`? use 20 seconds, and update the other 2  `context.WithTimeout` in this file to use the new constant. thanks :)

**Review by pacmessica — CHANGES_REQUESTED** [review]

**Comment by pacmessica** [comment]
let's merge this after [indexer manager pr](https://github.com/figment-networks/indexer-manager/pull/20) so that we can update go.mod :)

File: test/integration/api_test.go (hunk 2 above)
Comment (eesmerdag) [reply test/integration/api_test.go:298]: it is zero, no?

Comment (pacmessica) [reply test/integration/api_test.go:298]: yeaup

**Review by pacmessica — APPROVED** [review]

**Pull request merged**